### PR TITLE
Cone and lightning example

### PR DESCRIPTION
**Pull request by natascher**

### CODE REVIEW

**Comment by fabianmoronzirfas** [comment]
I'll review this when I'm back in Germany. Tuesday.

**Comment by fabianmoronzirfas** [comment]
Hi @natascher this code does not work for me. Some things I noted:  

- There is no draw function
- You have some global variables in between the functions. (This is a no go)
- You have some unused code
- To keep coding style consistent within this project you should add the `;` to the end of the lines.

File: motion/p5js/cones/index.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+function setup() {
+  createCanvas(300, 300, WEBGL)
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Please add `;` the line endings to keep the coding style consistent.

File: motion/p5js/cones/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  frameRate(30)
+}
+
+const height = 60
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Global variables go to the top of the script

File: motion/p5js/cones/index.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+const radius = 60
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Global variables go to the top of the script

File: motion/p5js/cones/index.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+function drawRotated(count, position) {
+  push()
+
+  rotateX(radians(90))
+  drawCones(count, position)
+
+  pop()
+}
+
+let y = 0
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Global variables go to the top of the script

File: motion/p5js/cones/index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+let increment = 1
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Global variables go to the top of the script

File: motion/p5js/cones/index.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+function draw_error() {
+  camera(0, 0, 300)
+  background(250)
+
+  // const locY = (mouseY / height - 0.5) * (-8);
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: unused code

File: motion/p5js/cones/index.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  // const locX = (mouseX / width - 0.5) * 8;
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: unused code

File: motion/p5js/cones/index.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+  drawRotated(5)
+
+  translate(2* radius, 2 * radius)
+
+  drawRotated(3)
+
+  translate(2 * radius, 2 * radius)
+
+  drawRotated(1)
+}
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: `function draw(){}` is fully missing.

File: motion/p5js/cones/index.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Natalie Schreiber
+author-url: https://github.com/natascher
+title: cones
+tags: [P5.js, motion, "Law of Simplicity"]
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.md]: Please remove the "" around the law name

File: motion/p5js/cones/index.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
+const height = 60;
+const radius = 60;
+let y = 0;
+let increment = 1;
+
+function setup() {
+  createCanvas(300, 300, WEBGL);
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Please make it 500 × 500 again

**Comment by fabianmoronzirfas** [comment]
Hi @natascher you are nearly there. See the requested changes

File: repeat/p5js/hidden-circles/index.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  distance = distance || 20;
+  
+  for (var i = 0; i < count; i++) {
+     var x = i * distance;
+    
+    line(x, 0, x, 500);
+  }
+}
+
+function draw() {
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: Please use DRY code. Don't repeat yourself. The ellipses below can be done in a single loop. I also would add the "Law of Closure" tag.

File: repeat/p5js/hidden-circles/index.js [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  stroke(0);
+  strokeWeight(2);
+  drawLines(100, dis);
+  
+  strokeWeight(10);
+  stroke(255, 255, 255);
+  noFill();
+  ellipse(250, 250, 100);
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: Please use DRY code. Don't repeat yourself. The ellipses below can be done in a single loop. I also would add the "Law of Closure" tag.

File: motion/p5js/cones/index.md [hunk 13]
```diff
@@ -3,7 +3,7 @@ layout: p5js
 author: Natalie Schreiber
 author-url: https://github.com/natascher
 title: cones
-tags: [P5.js, motion, "Law of Simplicity"]
+tags: [P5.js, motion, Law of Space through Light and Shadow]
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.md]: Currently there is only the tag for the "Law of Space" for the sake of the current state please change it to only that.

File: motion/p5js/cones/index.js [hunk 14]
```diff
@@ -4,7 +4,7 @@ let y = 0;
 let increment = 1;
 
 function setup() {
-  createCanvas(300, 300, WEBGL);
+  createCanvas(500, 500, WEBGL);
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: The canvas needs to be assigned to div. Please change the code to this:  

```
  var canvas = createCanvas(500, 500, WEBGL);
  canvas.parent('sketch');
```

This allows to display it directly in the site.

File: motion/p5js/cones/index.js [hunk 15]
```diff
@@ -0,0 +1,72 @@
+const height = 60;
+const radius = 60;
+let y = 0;
+let increment = 1;
+
+function setup() {
+  createCanvas(500, 500, WEBGL);
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: The canvas needs to be assigned to the sketch div. See the documentation on contributing p5js sketches https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#code

File: repeat/p5js/hidden-circles/index.js [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+  drawEllipsis(150);
+  drawEllipsis(240);
+  drawEllipsis(300);
+  drawEllipsis(340);
+  drawEllipsis(400);
+  drawEllipsis(440);
+  drawEllipsis(500);
+}
+
+ function mouseClicked(){
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: Please remove the image saving code.

File: repeat/p5js/hidden-circles/index.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+  stroke(0);
+  strokeWeight(2);
+  drawLines(100, dis);
+
+  drawEllipsis(100)
+
+  fill(255);
+  ellipse(250, 250, 35);
+
+  drawEllipsis(150);
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.js]: This could be done more DRY with a loop.

File: repeat/p5js/hidden-circles/index.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Natalie Schreiber
+author-url: https://github.com/natascher
+title: hidden_circles
```
Comment (fabianmoronzirfas) [inline repeat/p5js/hidden-circles/index.md]: A title should be "Title Case" without underscores and what not

File: motion/p5js/cones/index.js [hunk 19]
```diff
@@ -0,0 +1,72 @@
...
+  if (y === 0) {
+    increment = 1;
+  } else if (y === height) {
+    increment = -1;
+  }
+}
+
+function draw(){
+  pointLight(250, 250, 250, y * 8, y * 8, 0);
+
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Lots of unneeded whitespace.

File: motion/p5js/cones/index.js [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+  translate(0, -4 * radius);
+
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: Lots of unneeded whitespace.

File: motion/p5js/cones/index.js [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+  drawRotated(1);
+
+  translate(-2 * radius, 2 * radius);
+
```
Comment (fabianmoronzirfas) [inline motion/p5js/cones/index.js]: (Lots of unneeded whitespace.) and so on…

File: repeat/p5js/hidden-circles/index.js (hunk 17 above)
Comment (frontendphil) [inline repeat/p5js/hidden-circles/index.js]: DRY (https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) does not necessarily mean don't write the same thing a couple of times. Here for instance the `drawEllipsis` function can be seen as a sub-component which is used a couple of times. So the code is actually dry. Everything that is needed to draw an ellipsis is encapsulated in this method. Using it a couple of times does not violate that rule. 

Also I doubt that a loop like

```jsx
[150, 240, 300, 340, 400, 440, 500].forEach((height) => drawEllipsis(height))
``` 

Would make this code easier to understand / better maintainable.

**Comment by fabianmoronzirfas** [comment]
closing because it is old and it should be commited to #51

**Pull request closed without merging**